src/utils/truffle.ts

```diff
@@ -0,0 +1,126 @@
+import { Address, ChainURI, TransactionHash, ABI, Compiler } from 'ethpm/package';
+
+
+interface DeploymentData {
+  contract_type: string,
+  address: Address,
+  transaction: TransactionHash,
+}
+
+interface Network {
+  address: Address,
+  transactionHash: TransactionHash,
+}
+
+interface Artifact {
```

Comment:
You don't need to define this yourself! You can just do:

```typescript
import { ContractObject } from "@truffle/contract-schema/spec";
```

Reply:
oohh! Good to know. Do you think it's worthwhile to include truffle as a dependency here? Truffle's fairly large, and I'm thinking there's a lot of pieces that we can use from truffle - but if we're aiming for a lightweight library here then it might not be the best idea to require it as a dependency?? Also unsure if any circular dependency issues might come into play...

Reply:
going ahead to merge this for now so I have something to work against today - but more than happy to come back and make the changes if you think it's worth adding truffle as a dependency